### PR TITLE
update cloneset statefulset daemonset docs for kruise 1.2.0

### PR DESCRIPTION
**Pull request by veophi**
Signed-off-by: mingzhou.swx <mingzhou.swx@alibaba-inc.com>

- Add docs for PRs:
  - https://github.com/openkruise/kruise/pull/954
  -  https://github.com/openkruise/kruise/pull/963
  - https://github.com/openkruise/kruise/pull/979
  - https://github.com/openkruise/kruise/pull/980

### CODE REVIEW

File: docs/user-manuals/cloneset.md [hunk 1]
```diff
@@ -270,6 +273,10 @@ When `partition` is set during update:
 - If it is a number: `(replicas - partition)` number of pods will be updated with the new version.
 - If it is a percent: `(replicas * (100% - partition))` number of pods will be updated with the new version.
 
+**FEATURE STATE:** Kruise v1.2.0
+- If `partition` is a percent, and `partition < 100% && replicas > 1` , CloneSet will ensure **at least one pod** will be updated with the new version.
+- For convenience, just read `.status.expectedUpdatedReplicas` field if you don't know how to calculate the number of the pods that will be updated.
```
Comment (furykerry) [inline docs/user-manuals/cloneset.md:278]: > For convenience, just read `.status.expectedUpdatedReplicas` field if you don't know how to calculate the number of the pods that will be updated.

One can compare .status.expectedUpdatedReplicas and .status.expectedUpdatedReplicas to decide whether workload had finish rolling out new revision under partition restriction.

File: docs/user-manuals/advanceddaemonset.md [hunk 2]
```diff
@@ -217,6 +225,20 @@ metadata:
...
+**FEATURE STATE:** Kruise v1.2.0
+
+```yaml
+  lifecycle:
+    preDelete:
+      markPodNotReady: true
+      finalizersHandler:
+      - example.io/unready-blocker
+```
+If you set `markPodNotReady=true` for `preDelete`, Kruise will try to set `KruisePodReady` condition to `False` when Pods enter `PreparingDelete` lifecycle state, and Pods will be **NotReady**, but containers still `Running`.
```
Comment (furykerry) [inline docs/user-manuals/advanceddaemonset.md:238]: plz add comments about the use case, such as
One can use markPodNotReady to drain service traffic before terminate containers.

File: docs/user-manuals/cloneset.md (hunk 1 above)
Comment (veophi) [reply docs/user-manuals/cloneset.md]: done

**Review by FillZpp — APPROVED** [review]
/lgtm

**Pull request merged**